streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java

```diff
@@ -181,8 +210,93 @@ public void createTopic(final String topic,
         IntegrationTestUtils.waitForTopicPartitions(brokers(), topicPartitions, TOPIC_CREATION_TIMEOUT);
     }
 
-    public void deleteTopic(final String topic) {
-        brokers[0].deleteTopic(topic);
+    /**
+     * Deletes a topic returns immediately.
+     *
+     * @param topic the name of the topic
+     */
+    public void deleteTopic(final String topic) throws Exception {
+        deleteTopicsAndWait(-1L, topic);
+    }
+
+    /**
+     * Deletes a topic and blocks for max 30 sec until the topic got deleted.
+     *
+     * @param topic the name of the topic
+     */
+    public void deleteTopicAndWait(final String topic) throws Exception {
+        deleteTopicsAndWait(TOPIC_DELETION_TIMEOUT, topic);
+    }
+
+    /**
+     * Deletes a topic and blocks until the topic got deleted.
+     *
+     * @param timeoutMs the max time to wait for the topic to be deleted (does not block if {@code <= 0})
+     * @param topic the name of the topic
+     */
+    public void deleteTopicAndWait(final long timeoutMs, final String topic) throws Exception {
+        deleteTopicsAndWait(timeoutMs, topic);
+    }
+
```

Comment:
What if we removed the singular `deleteTopic` overloads and just went with the `deleteTopics` instead?  Just a suggestion, I don't have a strong opinion on this.